tests/queries/0_stateless/03406_dictionary_lifetime.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```

Comment:
Same comments as in the other test apply here.

Minor comment: Could we maybe rename the test to 03406_dictionary_incorrect_min_max_lifetimes?

Reply:
Let's rename the other test similarly: tests/queries/0_stateless/03405_ssd_cache_incorrect_min_max_lifetimes_and_block_size.sh